### PR TITLE
Expose HTTP/gRPC metrics to OpenTelemetry as well

### PR DESCRIPTION
**Pull request by dmathieu**
## Motivation/summary

This modifies the HTTP and gRPC metrics middlewares to expose data to OpenTelemetry as well, so we can expose those metrics to the local exporter.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [x] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

This is unit-tested. It can also be tested manually by running apm-server with Tilt and self instrumentation enabled.

### CODE REVIEW

File: internal/beater/middleware/monitoring_middleware.go [hunk 1]
```diff
@@ -18,36 +18,79 @@
...
 }
+
+func (m *monitoringMiddleware) inc(id request.ResultID) {
+	if counter, ok := m.ints[id]; ok {
+		counter.Inc()
+	}
+}
+
+func (m *monitoringMiddleware) getMetric(n request.ResultID) metric.Int64Counter {
+	name := "http.server." + n
```
Comment (dmathieu) [inline internal/beater/middleware/monitoring_middleware.go:72]: I named those metrics `http.server.xxx`, and not `apm-server.http.server.xxx`, as the former better matches the otel semantic conventions.
https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md

We can rename those metrics in the local exporter.

**Review by simitt — COMMENTED** [review]
Didn't dive too deep into the technical details, but overall approach LGTM

File: internal/beater/interceptors/metrics_test.go [hunk 2]
```diff
@@ -76,8 +88,14 @@ func TestMetrics(t *testing.T) {
 				request.IDResponseErrorsTimeout:      0,
 				request.IDResponseErrorsUnauthorized: 0,
 			},
+			expectedOtel: map[string]int64{
+				"grpc.server." + string(request.IDRequestCount):        1,
+				"grpc.server." + string(request.IDResponseCount):       1,
+				"grpc.server." + string(request.IDResponseErrorsCount): 1,
```
[the remark this answers is not in the document]
Comment (dmathieu) [reply internal/beater/interceptors/metrics_test.go:93]: I have removed the metric from monitoring.Int, and the test doesn't fail either. So it does seem it's not being incremented in that interceptor (and the monitoring.Int test doesn't check for metrics that aren't emitted).

**Comment by lahsivjar** [comment]
Tested via https://github.com/elastic/apm-server/pull/11582#issuecomment-1776606521

**Pull request merged**